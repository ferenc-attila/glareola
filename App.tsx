import React, { useState } from 'react';
import { Pressable, StatusBar, StyleSheet, Text, View } from 'react-native';
import 'reflect-metadata';
import { FontAwesome5 } from '@expo/vector-icons';
import { Main } from './src/components/Main';
import { About } from './src/components/About';
import { ObservationList } from './src/components/ObservationList';
import { Settings } from './src/components/Settings';

export default function App() {

    const [appScreen, setAppScreen] = useState('main')

    function getScreen(appScreen: string) {
        switch (appScreen) {
            case ('main'): {
                return <Main></Main>
            }
            case ('about'): {
                return <About></About>
            }
            case ('observations'): {
                return <ObservationList></ObservationList>
            }
            case ('settings'): {
                return <Settings></Settings>
            }
        }
    }

    return (
        <>
            <StatusBar/>
            <View style={styles.buttonContainer}>
                <View style={styles.button}>
                    <Pressable onPress={() => setAppScreen('main')}>
                        <FontAwesome5 name='home' size={24} color='white' />
                    </Pressable>
                </View>
                <View style={styles.button}>
                    <Pressable onPress={() => setAppScreen('observations')}>
                        <FontAwesome5 name='binoculars' size={24} color='white' />
                    </Pressable>
                </View>
                <View style={styles.button}>
                    <Pressable onPress={() => setAppScreen('settings')}>
                        <FontAwesome5 name='tools' size={24} color="white" />
                    </Pressable>
                </View>
                <View style={styles.button}>
                    <Pressable onPress={() => setAppScreen('about')}>
                        <FontAwesome5 name="info-circle" size={24} color="white" />
                    </Pressable>
                </View>
            </View>
            {getScreen(appScreen)}
        </>
    );
}

const styles = StyleSheet.create({
    buttonContainer: {
        padding: 25,
        flexDirection: 'row',
        flexWrap: 'wrap',
        justifyContent: 'space-around',
        alignItems: 'center',
        backgroundColor: '#2e3a24'
    },
    button: {
        padding: 10,
        borderRadius: 5,
        backgroundColor: '#4d9460',
        color: '#ffffff',
        fontSize: 15,
        fontWeight: 'bold'
    }
});
