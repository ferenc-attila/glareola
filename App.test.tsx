import { waitFor } from '@testing-library/react-native';
import { fireEvent, render } from '@testing-library/react-native/build/pure';

import App from './App';

describe('App', () => {
    test('should render correctly', () => {
        const container = render(<App />);
        expect(container.toJSON).toMatchSnapshot();
    });

    describe('navigation', () => {
        test('loading the Home screen by default', () => {
            const container = render(<App />);
            expect(container.getByText('Glareola')).toBeTruthy();
            expect(container.getByText('A crawler for rarity observations')).toBeTruthy();
        });

        test('loading the About screen when about button pressed', async () => {
            const container = render(<App />);
            const aboutButton = container.getByLabelText('About button');
            fireEvent.press(aboutButton);
            expect(container.getByText('Cover Image')).toBeTruthy();
        });

        test.skip('loading the Observations screen when observations button pressed', async () => {
            // TODO: Refactor the code so that the url could come from settings, or .env file. We need state management to do this.
            const container = render(<App />);
            const observationsButton = container.getByLabelText('Observations button');
            await waitFor(() => fireEvent.press(observationsButton), { timeout: 3000 });
            expect(container.getByText('Observations')).toBeTruthy();
        });

        test('loading the Settings screen when setting button pressed', async () => {
            const container = render(<App />);
            const settingsButton = container.getByLabelText('Settings button');
            fireEvent.press(settingsButton);
            expect(container.getByText('Settings')).toBeTruthy();
        });

        test('loading the Home screen when home button pressed in Settings screen', async () => {
            const container = render(<App />);
            const settingsButton = container.getByLabelText('Settings button');
            fireEvent.press(settingsButton);
            const homeButton = container.getByLabelText('Home button');
            fireEvent.press(homeButton);
            expect(container.getByText('Glareola')).toBeTruthy();
            expect(container.getByText('A crawler for rarity observations')).toBeTruthy();
        });
    });

    describe('buttons', () => {
        test('should have correct accessibility label', () => {
            const container = render(<App />);
            expect(container.getByLabelText('Home button')).toBeTruthy();
            expect(container.getByLabelText('About button')).toBeTruthy();
            expect(container.getByLabelText('Observations button')).toBeTruthy();
            expect(container.getByLabelText('Settings button')).toBeTruthy();
        });
    });
});
