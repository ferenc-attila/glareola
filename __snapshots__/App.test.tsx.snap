// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> App renders correctly 1`] = `
[
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2e3a24",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "space-around",
        "padding": 25,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#4d9460",
          "borderRadius": 5,
          "color": "#ffffff",
          "fontSize": 15,
          "fontWeight": "bold",
          "padding": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text />
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#4d9460",
          "borderRadius": 5,
          "color": "#ffffff",
          "fontSize": 15,
          "fontWeight": "bold",
          "padding": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text />
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#4d9460",
          "borderRadius": 5,
          "color": "#ffffff",
          "fontSize": 15,
          "fontWeight": "bold",
          "padding": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text />
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#4d9460",
          "borderRadius": 5,
          "color": "#ffffff",
          "fontSize": 15,
          "fontWeight": "bold",
          "padding": 10,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text />
      </View>
    </View>
  </View>,
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2e3a24",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 40,
          "fontWeight": "900",
          "paddingBottom": 25,
        }
      }
    >
      Glareola
    </Text>
    <Image
      source={1}
      style={
        {
          "margin": 30,
          "maxHeight": "50%",
          "maxWidth": "90%",
          "resizeMode": "contain",
        }
      }
    />
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 22,
          "fontWeight": "500",
          "textAlign": "center",
          "width": "90%",
        }
      }
    >
      A crawler for rarity observations
    </Text>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 10,
          "marginTop": 25,
        }
      }
    >
      0.0.5
    </Text>
  </View>,
]
`;
